File: lua/checkduplicate.lua
function CheckDuplicates(nums)
	local hash_set = {}
	for _, number in ipairs(nums) do
		if hash_set[number] then
			return true
		end
		hash_set[number] = true
	end
	return false
end

local dupes = { 1, 1, 2, 3 } -- prints 'stinky butt'
local years = { 2011, 2012, 1990 } -- prints 'falshy'
local containsDupe = CheckDuplicates(dupes)

if containsDupe then
	print("stinky butt")
else
	print("falshy")
end

File: cpp/checkduplicate.cpp
#include <format>
#include <iostream>
#include <unordered_set>
#include <vector>

bool CheckDuplicates(std::vector<int> nums)
{
	std::unordered_set<int> hash_set{};
	int i = 0;
	while (i < nums.size())
	{
		if(auto iter = hash_set.find(nums[i]); iter != hash_set.end())
		{
			return true;
		}
		hash_set.insert(nums[i]);
		++i;
	}
	return false;
}

int main()
{
	std::vector<int> dupenums{1, 1, 2, 2, 3 ,3}; // prints 'stinky dupe'
	std::vector<int> nums{1, 2, 3}; // prints 'falshy'
	bool dupeFound = CheckDuplicates(dupenums);
	if (dupeFound)
	{
		std::cout << "stinky dupe\n";
		// std::format("stinky dupe\n");
	}
	else 
	{
		std::cout << "falshy\n";
		// std::format("falshy\n");
	}
	return 0;
}
